### PR TITLE
Add semver validator integration test suite

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
$subject.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/37033
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/37528

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/SemverChecker.java [hunk 1]
```diff
@@ -109,12 +100,12 @@ public Optional<PackageDiff> computeDiff() throws SemverToolException {
         String pkgName = currentPackage.packageName().value();
         SemanticVersion pkgVersion = currentPackage.packageVersion().value();
 
-        CentralClientWrapper clientWrapper = new CentralClientWrapper(outStream, errStream);
-        if (releasedVersion == null) {
-            errStream.println("checking for the latest compatible release version available in central...");
-            releasedVersion = clientWrapper.getLatestCompatibleVersion(orgName, pkgName, pkgVersion);
+        BallerinaPackageResolver pkgResolver = new BallerinaPackageResolver(outStream, errStream);
```
Comment (lnash94) [inline misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/SemverChecker.java:103]: I'm curious whether we use `errStream` to print errors?

File: tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/semver/SemverValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+        List<String> expectedLogs = new LinkedList<>();
+        expectedLogs.add("current version: 2.0.0");
+        expectedLogs.add("compared with the release version '1.0.0'");
+        expectedLogs.add("backward-incompatible changes detected");
+        expectedLogs.add("suggested version: 2.0.0");
+        // diff output related logs
+        expectedLogs.add("Comparing version '2.0.0'(local) with version '1.0.0'(central)");
+        expectedLogs.add("[+-] package 'library' is modified [version impact: MAJOR]");
+        expectedLogs.add("  [+-] module 'library' is modified [version impact: MAJOR]");
```
Comment (lnash94) [inline tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/semver/SemverValidatorTest.java:116]: I'm wonder whether there is a chance to change the white space, and tab space according to the OS and will it matter for test comparison?

File: misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/SemverChecker.java (hunk 1 above)
Comment (NipunaRanasinghe) [reply misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/SemverChecker.java:103]: Yes we use it to output the errors.

File: tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/semver/SemverValidatorTest.java (hunk 2 above)
Comment (NipunaRanasinghe) [reply tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/semver/SemverValidatorTest.java:116]: IMO we shouldn't be having issues as here we only have spaces. This test is passing in both windows and linux builds

Comment (lnash94) [reply tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/semver/SemverValidatorTest.java:116]: Then it would be fine :)

**Review by AzeemMuzammil — DISMISSED** [review]
LGTM

**Review by lnash94 — DISMISSED** [review]

**Review by lnash94 — APPROVED** [review]

**Pull request merged**